styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -160,13 +160,12 @@ void tick() {
     timedOutInstances.forEach(this::sendTimeout);
 
     for (InstanceState eligibleInstance : eligibleInstances) {
-      final boolean proceed = limitAndDequeue(
-          resources, workflowResourceReferences, currentResourceUsage, eligibleInstance);
-      if (!proceed) {
+      if (!dequeueRateLimiter.tryAcquire()) {
```

Comment:
Why is the dequeue rate limiting moved here? It seems undesirable to count e.g. dequeues that didn't happen because they were resource limited towards the rate limit.

Reply:
Good point, I misinterpreted how the RateLimiter works. Reverting the commit.